### PR TITLE
feat(core): add wishlist details

### PR DESCRIPTION
**Pull request by yurytut1993**
## What/Why?
This pr adds Wishlist Details Page to Account Wishlist and fix minor translation issues for Wishlist Book

## Testing
Locally

### CODE REVIEW

File: core/components/product-card/delete-wishlist-item-form/index.tsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+        const timer = setTimeout(() => {
+          toast.dismiss(toastId);
+
+          onWishlistItemDeleted(itemEntityId);
+
+          void deleteWishlistItem({
+            input: { entityId, itemEntityIds: [itemEntityId] },
+          });
+        }, 5000);
```
Comment (yurytut1993) [inline core/components/product-card/delete-wishlist-item-form/index.tsx:60]: This is workaround for current [react-hot-toast](https://www.npmjs.com/package/react-hot-toast) library we use. In future we may change it to [react-toastify](https://www.npmjs.com/package/react-toastify) that has useful onClose callback needed to pass deleteWishlistItem mutation in

File: core/app/[locale]/(default)/account/(tabs)/wishlists/_components/update-wishlist-form/_actions/update-wishlist.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  name: z.string(),
+});
+
+export const updateWishlist = async (formData: FormData) => {
+  const { entityId, name } = UpdateWishlistSchema.parse({
+    entityId: formData.get('entityId'),
+    name: formData.get('name'),
+  });
+
+  const input = {
```
Comment (bc-alexsaiannyi) [inline core/app/[locale]/(default)/account/(tabs)/wishlists/_components/update-wishlist-form/_actions/update-wishlist.ts:18]: just thought to consider, if we don't reuse input variable else where maybe we remove it and define arguments directly on L26?🤔

File: core/client/mutations/update-wishlist.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
+import { getSessionCustomerId } from '~/auth';
```
Comment (jorgemoya) [inline core/client/mutations/update-wishlist.ts:1]: We no longer define mutation in this folder unless shared across multiple instances. Seems to me like you can simply use as needed in your form action.

File: core/client/mutations/update-wishlist.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+import { client } from '..';
+import { graphql, VariablesOf } from '../graphql';
+
+const UPDATE_WISHLIST_MUTATION = graphql(`
```
Comment (jorgemoya) [inline core/client/mutations/update-wishlist.ts:6]:
```suggestion
const UpdateWishlistMutation = graphql(`
```

File: core/app/[locale]/(default)/account/(tabs)/_components/modal.tsx [hunk 5]
```diff
@@ -9,7 +9,7 @@ import { Button } from '~/components/ui/button';
 interface Props extends PropsWithChildren {
   actionHandler?: MouseEventHandler<HTMLButtonElement>;
   title: string;
-  trigger: React.ReactNode;
+  trigger?: React.ReactNode;
```
Comment (jorgemoya) [inline core/app/[locale]/(default)/account/(tabs)/_components/modal.tsx:12]: Don't you get an error/warning if there's no trigger defined for a modal using radix?

File: core/app/[locale]/(default)/account/(tabs)/wishlists/_components/update-wishlist-form/index.tsx [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+  FieldLabel,
+  FieldMessage,
+  Form,
+  FormSubmit,
+  Input,
+} from '~/components/ui/form';
+
+import { updateWishlist } from './_actions/update-wishlist';
+
+type Wishlist = NonNullable<Awaited<ReturnType<typeof createWishlistMutation>>>;
```
Comment (jorgemoya) [inline core/app/[locale]/(default)/account/(tabs)/wishlists/_components/update-wishlist-form/index.tsx:20]: We have an `ExistingResultType` util you could use here I believe.

File: core/app/[locale]/(default)/account/(tabs)/wishlists/_components/wishlist-details.tsx [hunk 7]
```diff
@@ -0,0 +1,205 @@
...
+import { useRouter } from '~/i18n/routing';
+
+import { useAccountStatusContext } from '../../_components/account-status-provider';
+import { Modal } from '../../_components/modal';
+import { getWishlist } from '../page-data';
+
+import { DeleteWishlistForm } from './delete-wishlist-form';
+import { UpdateWishlistForm } from './update-wishlist-form';
+
+type Wishlist = NonNullable<Awaited<ReturnType<typeof getWishlist>>>['wishlists'][number];
```
Comment (jorgemoya) [inline core/app/[locale]/(default)/account/(tabs)/wishlists/_components/wishlist-details.tsx:21]: Same here for use of util.

File: core/client/mutations/update-wishlist.ts (hunk 3 above)
Comment (yurytut1993) [reply core/client/mutations/update-wishlist.ts:1]: I plan to create small pr with refactoring for all wishlist mutations in nearest time

File: core/client/mutations/update-wishlist.ts (hunk 4 above)
Comment (yurytut1993) [reply core/client/mutations/update-wishlist.ts:6]: I will use your suggestion in next pr where all wishlist mutations. For now prefer to use current name for it to be consistent with other mutations

File: core/app/[locale]/(default)/account/(tabs)/_components/modal.tsx (hunk 5 above)
Comment (yurytut1993) [reply core/app/[locale]/(default)/account/(tabs)/_components/modal.tsx:12]: no, no errors

**Pull request merged**